clusters/script-gen-readonly-kubeconfig.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+cat <<EOF | kubectl apply -f -
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: crole-pod-readyonly
+rules:
+- apiGroups: [""]
+  resources: ["pods"]
+  verbs: ["get", "watch", "list"]
```

Comment:
the original proposal was cluster wide pod only
here @nikimanoledaki has a say

Reply:
Thank you @maxbischoff & @dipankardas011 for looking into this! The only strict rule was to exclude Secrets, which the default [`view` ClusterRole](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#user-facing-roles) does as well. I didn't know that there were default out-of-the-box Roles. :)

Limiting to `pod` was so that we can start small if we were to create a Role from scratch. However, if we can benefit from the ease of having a default role out of the box that already excludes Secrets, I don't see any reason against this - @antoniodituri @rossf7 wdyt?

Reply:
@dipankardas011 did you have a chance to test the script with the default `view` ClusterRole?

Reply:
Okay will check it out
But one question that do we actually need it?
Should the user have read access to all the other resources except secret is it good ?

Reply:
I'm late to the party but +1 from me on using the default view ClusterRole.

I think secrets are the only resources we need to restrict. Being able to view the rest of the resources including the Flux resources will be useful for trouble shooting.

Reply:
got it 👍🏼

Reply:
@rossf7 the default `view` ClusterRole does not grant read access to custom resources, such as Flux resources. It only grants access to core kubernetes resources. You can check this by looking at the ClusterRole using `kubectl get clusterrole view -o yaml`

We'll have to add a second ClusterRole and bind it to grant access to Custom Resources.

Reply:
@maxbischoff TIL 😄 thank you and I think a second ClusterRole for the Flux custom resources would be useful.

Reply:
added aggregation support